liblangutil/CharStream.cpp

```diff
@@ -144,3 +144,31 @@ string CharStream::singleLineSnippet(string const& _sourceCode, SourceLocation c
 
 	return cut;
 }
+
+optional<int> CharStream::translateLineColumnToPosition(int _line, int _column) const
+{
+	return translateLineColumnToPosition(m_source, _line, _column);
+}
+
+optional<int> CharStream::translateLineColumnToPosition(std::string const& _text, int _line, int _column)
+{
+	size_t offset = 0;
+
+	while (_line > 0)
+	{
+		auto const index = _text.find('\n', offset);
+		if (index != _text.npos)
+		{
+			offset = index + 1;
+			_line--;
+		}
+		else
+			return nullopt;
```

Comment:
This algorithm always returns nullopt if the source contains a single line without `\n` at the end. I don't think this is how it should be.

Reply:
added tests including for that case.